assets/app/scripts/services/data.js

```diff
@@ -650,9 +649,14 @@ angular.module('openshiftConsole')
       $rootScope.$apply(function() {
         self._watchCallbacks(type, context).fire(self._data(type, context), eventData.type, eventData.object);
       });
+
+      // Reset the retries count once we've successfully processed a message.
+      // Wait until this point to prevent continuously looping if there are any errors processing a response.
+      this._watchWebsocketRetries(type, context, 0);
```

Comment:
I might put this after the resourceVersion/data update, before calling callbacks